src/System.Buffers.Experimental/System/Range.cs

```diff
@@ -13,8 +13,9 @@ public struct Range : IEnumerable<int>
         public const int UnboundedLast = Int32.MaxValue;
 
         public readonly int First;
-        public readonly uint Length;
-        public int Last => (int)(Length + First - 1);
+        public readonly int Last;
+
+        public uint Length => (uint)((long)Last - (long)First);
```

Comment:
This is going to force consumers of Length to cast to int right?  Say if they need to pass Range.Length into String.IndexOf or similar methods that take a Length/count parameter.